[None][feat] Export ONNX for DriveOS LLM

nvyocox opened the pull request:
## Description

### The issue

To run inference on DriveOS LLM efficiently, the graph must be optimized. Currently the workflow of DriveOS LLM is rewrite the graph in ONNX level. However, matching and manipulation on ONNX is difficult and tedious because is has been converted to fine-grained ops.

### The solution

In this PR, we utilize TensorRT LLM AutoDeploy. AutoDeploy already done lots of optimization, and we only have to adapt the optimized graph to feat DriveOS LLM's requirements and export the result graph to checkpoint with ONNX format.

The original workflow:

```
HF model ------------------> ONNX ---> TRT Engine
             DriveOS LLM
```
The new workflow

```
HF model ---> FX Graph ------------------ FX Graph -----> ONNX ---> TRT Engine
                       AutoDeploy rewrite          export
```

We add a new AutoDeploy configuration `export_driveos_llm_onnx` to do that. This configuration is like the default configuration with follow difference:

- It does not rewrite the attention to KV cached attention.
- It fuse the rope & attention ops into DriveOS LLM's `AttentionPlugin` op.
- It export the result graph into ONNX.
- It add new graph output for each attention, return the new KV cache result.
- It add some new input for DriveOS LLM's `AttentionPlugin`.

### Visualization

During the development we also add a debugging feature to visualize the FX Graph after each transformation. See issue https://github.com/NVIDIA/TensorRT-LLM/issues/9370

## Test Coverage

This PR has 2 tests:

1. Test transformation to fuse the rope & attention op.
2. End-to-end test on Qwen2.5-0.5B model

To run the test

```shell
cd tests
# Fuse unit test
pytest unittest/_torch/auto_deploy/unit/singlegpu/transformations/library/test_fuse_rope_attention.py
# End to end test
pytest unittest/_torch/auto_deploy/unit/singlegpu/test_ad_export_onnx.py
```

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

* **New Features**
  * Added support for exporting LLM models to ONNX format for DriveOS deployment with automatic configuration and tokenizer export.
  * Added graph visualization capabilities to analyze model architecture as Graphviz diagrams.
  * Added chat template processing for vision-language and text-only models.

* **Tests**
  * Added ONNX export and RoPE attention fusion transformation tests.

* **Chores**
  * Updated dependencies and Docker base image configuration.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

<!--
Please write the PR title by following this template:

**[JIRA ticket/NVBugs ID/GitHub issue/None][type] Summary**

Valid ticket formats:
  - JIRA ticket: [TRTLLM-1234] or [FOOBAR-123] for other FOOBAR project
  - NVBugs ID: [https://nvbugs/1234567]
  - GitHub issue: [#1234]
  - No ticket: [None]

Valid types (lowercase): [fix], [feat], [doc], [infra], [chore], etc.

Examples:
  - [TRTLLM-1234][feat] Add new feature
  - [https://nvbugs/1234567][fix] Fix some bugs
  - [#1234][doc] Update documentation
  - [None][chore] Minor clean-up

Alternative (faster) way using CodeRabbit AI:

**[JIRA ticket/NVBugs ID/GitHub issue/None] @coderabbitai title**

NOTE: "@coderabbitai title" will be replaced by the title generated by CodeRabbit AI, that includes the "[type]" and title.
For more info, see /.coderabbit.yaml.

-->

## Description

<!--
Please explain the issue and the solution in short.
-->

## Test Coverage

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.


- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--reuse-test (optional)pipeline-id --disable-fail-fast --skip-test --stage-list "A10-PyTorch-1, xxx" --gpu-type "A30, H100_PCIe" --test-backend "pytorch, cpp" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx" --detailed-log --debug(experimental)]`

Launch build/test pipelines. All previously running jobs will be killed.

`--reuse-test (optional)pipeline-id ` *(OPTIONAL)* : Allow the new pipeline to reuse build artifacts and skip successful test stages from a specified pipeline or the last pipeline if no pipeline-id is indicated. If the Git commit ID has changed, this option will be always ignored. The DEFAULT behavior of the bot is to reuse build artifacts and successful test results from the last pipeline.

`--disable-reuse-test ` *(OPTIONAL)* : Explicitly prevent the pipeline from reusing build artifacts and skipping successful test stages from a previous pipeline. Ensure that all builds and tests are run regardless of previous successes.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-PyTorch-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-PyTorch-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--test-backend "pytorch, cpp"` *(OPTIONAL)* : Skip test stages which don't match the specified backends. Only support [pytorch, cpp, tensorrt, triton]. Examples: "pytorch, cpp" (does not run test stages with tensorrt or triton backend). Note: Does **NOT** update GitHub pipeline status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests in addition to running L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx".

`--detailed-log ` *(OPTIONAL)* : Enable flushing out all logs to the Jenkins console. This will significantly increase the log volume and may slow down the job.

`--debug ` *(OPTIONAL)* : **Experimental feature**. Enable access to the CI container for debugging purpose. Note: Specify exactly one stage in the `stage-list` parameter to access the appropriate container environment. Note: Does **NOT** update GitHub check status.

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`
and the `scripts/test_to_stage_mapping.py` helper.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

nvyocox committed:
[None][feat] AutoDeploy ONNX export

[none][feat] Add AutoDeploy export-onnx mode

Add a new mode "export-onnx" to AutoDeploy.
The mode is almost identical to the default one with 2 difference:

  1. Fuse torch_rope_with_explicit_cos_sin &
     torch_cached_attention_with_cache into onnx_rope_attnetion
  2. The result is not TRT Engine but .onnx

Files added:

  - export_onnx.py: The transformation to fuse the ops
  - graph_module_visualizer.py: Convert GraphModule to .dot
  - examples/onnx_export_llm.py: Example usage
  - onnx_driveos_llm.yaml: The new mode config file
  - onnx_attnetion.py: The definition of the fused op

[none][feat] fix small graphviz bug, remove useless code

[none][feat] Rename mode from onnx_driveos_llm to export_driveos_llm_onnx

[none][feat] Rename export_onnx.py to fuse_rope_attention.py

[none][feat] Annotate .meta['val'] with add_graph_input()

[none][feat] Successfully export .onnx

[none][feat] Add set_kvcache_placeholder_metadata transform

[none][feat] Skip torch_cached_attention_prepare_metadata

[none][feat] Fix SetKVCachePlaceholderMetadata transform

[none][feat] Remove unused placeholder of prepare_metadata

[none][feat] Fix to run DeepSeek-R1

[none][feat] Add remove_graph_input, refactor remove_unused_placeholder()

[none][feat] Merge K&V cache placeholder

[none][feat] Replace sin_cos with input

[none][feat] Manually fuse rope & attn

[none][feat] Export torch_attention_bsnd_grouped_sdpa with dynamic shape

[none][feat] Manually match rope & attn, not replace yet

[none][feat] Successfully export ONNX with dynamic input

[none][feat] Hack out_spec to add graph output

[none][feat] Fix present_key_values shape

[none][feat] Fix input & output names

[none][feat] Change out_spec in add_graph_output

[none][feat] Fix export of torch_linear_simple

The original translation misses a transpose on the weight.

[none][feat] Fix present_key_values shape

[none][feat] Rewire reshape's new shape as TRT-LLM edge

[none][feat] Fix non-text rebase conflicts

[none][feat] Fix AttentionPlugin domain. should be "" not "ai.onnx"

[none][feat] Enhance visualize, use .meta["val"] instead of .meta["tensor_meta"]

[none][feat] Fix visualize tensor width calculation

When calculate the width of the tensor, check it the dimension is a int or SymInt.

The original implementation accidentally introduce constraints to the symbol int.
I don't execlty know how it happen. actually I don't think it should
introduce new constraints, but it dose.

[none][feat] Fix output dynamic batch_size

Originally max batch size is 2, however, don't know why, when set to 2,the batch_size will collapse to literal static int 2 even we explicitly it is dynamic axis.
And more weird, when set to 13, the batch_size will be dynamic.
default=13,  # to enable dynamic batch_size, the match size must > 1

[none][feat] Rename fuse_rope_attention_manually to fuse_rope_attention

[none][feat] Remove fuse_rope_attention.py

[none][feat] Rewire reshape to make the graph like Luxiao's

[none][feat] Fix last_token_ids dtype from i32 to i64

[none][feat] Catch up update to date DriveOS LLM

- Add placeholder kvcache_start_index
- AttentionPlugin add input kvcache_start_index
- Insert Unsqueeze -1 before GatherND
- rope_rotary_cos_sin dynamic axis name changed from
  rope_max_position_length to max_position_embeddings
- logits' dtype should be float32, insert a cast
- Insert cast to f16 before AttentionPlugin
- All cast to bf16 should be f16

[none][feat] Catch up update to date DriveOS LLM

- model.half() convert whole model to f16, including weight
- Remove AttentionPlugin attribute kv_cache_capacity & max_batch_size
- AttentionPlugin output[1] shape infer by seq_len + past_len
- AttentionPlugin domain changed from `onnx.ai` to `trt`
- Placeholder `kvcache_start_index` dynamic axes changed from `batch_size` to `kv_cache_start_batch_size`

[none][feat] Catch up-to-date main

[none][feat] Add test for fuse_rope_attention transform

- Add test for fuse_rope_attention
- Enhance run_test_transformed_gm support Module with multiple input
- Fix add_graph_output for graph with only one _LEAF_SPEC

[none][feat] Add unit test for fuse_rope_attn

- Add a unit test
- Fix add_graph_output when out_spec is _LEAF_SPEC

[none][feat] Export .json files

[none][feat] add AutoDeploy export onnx end-to-end test

[none][feat] Export ONNX with cpu to reduce GPU memory footprint

[none][feat] Use model.config to get head_dim, instead of using literal

[none][feat] Visualize graph only when env var AD_DEBUG_VISUALIZE_DIR is set

- Now we don't visualize by default, only when AD_DEBUG_VISUALIZE_DIR is set.
- Also, AD_DEBUG_VISUALIZE_DIR is the output dir, so you can specify the output dir
- Simplify the logging message, move lots of msg to debug instead of info
- Add .cursor to .gitignore

Signed-off-by: yocox <yocox@nvidia.com>

nvyocox commented on tensorrt_llm/_torch/auto_deploy/transform/library/fuse_rope_attention.py:
```diff
@@ -0,0 +1,434 @@
+            with graph.inserting_before(match.attn_node):
+                cast_node = graph.call_function(
+                    torch.ops.aten.to.dtype, args=(qkv_node, torch.float16)
+                )
```
This is intentional, the DriveOS LLM expecting a float16 dtype

nvyocox committed:
[None][feat] Update per pohan's review

Signed-off-by: yocox <yocox@nvidia.com>

lucaslie reviewed (commented):
Looks great overall!

nvyocox commented on tensorrt_llm/_torch/auto_deploy/transform/library/gather_last_token_ids.py:
Agree, they are similar, but the semantic are different (for now). I had took a look of this transformation. The gather op in this transform has 3 inputs (2 from new input placeholder), but in DriveOS LLM's gather, it's 2 (1 from new input placeholder, the name of the placeholder is different, too).

I think there is a chance to unify those two :D I will survey that.

nvyocox commented on tensorrt_llm/_torch/auto_deploy/config/export_driveos_llm_onnx.yaml:
Thanks to point in out. I will enable it. Since in the AD configuration the model is compiled on CPU.

nvyocox replied:
It makes non-strict ONNX exporting failed. strict ONNX export still can success. Interesting, back to this issue later, to what actually done by `fuse_gemm`.

nvyocox commented on tensorrt_llm/_torch/auto_deploy/config/export_driveos_llm_onnx.yaml:
Works like a charm

nvyocox commented on tensorrt_llm/_torch/auto_deploy/transform/library/fuse_rope_attention.py:
Yes. I chose to use the pattern matching library first as well, and following are problems (some of them are fixed) I had encountered when I try to use our TorchInductory-based pattern match & rewrite library

- Previous transform insert new placeholder without annotating `.meta["val"]`, so following transform can not run pattern tracing on the graph. The `.meta["val"]` required by TorchInductor.
- TorchInductor requires the input type of the pattern must be a `torch.Tensor`, but, at that time, my pattern has scalar as input (the dimension of some tensor, like `t.size(2)`) because I want to make my result graph as close as possible to the DriveOS LLM's graph. Finally I found the ONNX exporter will rewrite the graph, too, and it modified some changed I have done. So I give up this requirement. 
- For each fused RoPE & attention, we need to insert a new placeholder and  a new output. However TorchInductor does not support modifying the input and output of the graph because it want to keep the interface of the graph. The concern is reasonable, but in that case, we have to separate the transformation into 3 steps:
  1. Add new input placeholders manually with `add_graph_input()`
  2. Fuse rope & attn to a new op with the TorchInductor tool
  3. Add new graph output manually with `add_graph_output()`

  It will work after all, but, see next reason.
- I encountered an issue to export ONNX with dynamic batch_size axis. Everything looks ok but the dynamic axis collapse from a symbol into a literal `2` when exporting ONNX. I really can not find the root cause, maybe the pattern I wrote is wrong. I ask  @nvchenghaoz help me to debug that. At the same time, I try the manually match & rewrite approach. And before chenghao solve that I got it works. By the way, my finally solution is very strange, I changed the example input batch size from `2` to `13` and it works, I am really confused.

And other issues I don't remember the details :(

At that time, because I had already spend too much time on the TorchInductor approach, I decided to stop and use the manually approach. But still I am very curious about why it did not work. Probably is the `2` vs `13`. Maybe in the future I can try it one more time :joy:

nvyocox commented on tensorrt_llm/_torch/auto_deploy/utils/_graph.py:
OK, I got it :D

nvyocox committed:
[none][feat] Update code per lucas's review

Signed-off-by: yocox <yocox@nvidia.com>

nvyocox commented on tensorrt_llm/_torch/auto_deploy/utils/_graph.py:
drop it :D

lucaslie commented on tensorrt_llm/_torch/auto_deploy/transform/library/gather_last_token_ids.py:
You shouldn't call this explicitly yourself. Instead you can do the following
```
TransformInfo(skipped=False, num_matches=1, is_clean=False, has_valid_shapes=False)
```

and then configure in the transform config to 
```
gather_last_token_ids:
    stage: export_onnx
    run_shape_prop: True
```

lucaslie commented on tensorrt_llm/_torch/auto_deploy/custom_ops/onnx_attention.py:
maybe a more descriptive name would be helpful. E.g.
```

```suggestion
@torch.library.custom_op("auto_deploy::torch_onnx_attention_plugin", mutates_args=())
```

lucaslie commented on tensorrt_llm/_torch/auto_deploy/transform/library/fuse_rope_attention.py:
is this needed here? Otherwise, you can rely on the cleanup logic at the end of the transform (same as other comments)

nvyocox replied:
You are right, it is not necessary. Thanks.

lucaslie commented on tensorrt_llm/_torch/auto_deploy/transform/library/fuse_rope_attention.py:
Same as other comments

lucaslie commented on tensorrt_llm/_torch/auto_deploy/transform/library/fuse_rope_attention.py:
consider using the `extract_op_args` utility

lucaslie commented on tensorrt_llm/_torch/auto_deploy/transform/library/fuse_rope_attention.py:
ditto

nvpohanh commented on docs/source/torch/auto_deploy/advanced/export_onnx.md:
does `TinyLlama/TinyLlama-1.1B-Chat-v1.0` work? Or should we use Qwen2.5 0.5b as the example?

nvyocox replied:
You are right, the TinyLlama is not work. I will change it to DeepSeek.

nvpohanh commented on docs/source/torch/auto_deploy/advanced/export_onnx.md:
change the default model?

nvpohanh commented on docs/source/torch/auto_deploy/advanced/export_onnx.md:
Remove `--max_seq_len` and `--max_batch_size` because they do not actually affect the dynamic ranges of the exported ONNX I/O tensor dims

nvpohanh commented on docs/source/torch/auto_deploy/advanced/export_onnx.md:
the returned object `llm` cannot be used, right? if so, change this to:

```
LLM(**ad_config.to_llm_kwargs())
```

nvyocox committed:
[none][feat] Update code per pohan & lucas's review

Signed-off-by: yocox <yocox@nvidia.com>

nvyocox committed:
[None][feat] Fix past_key_value_cache_XX axis 2

The dimension is wrong, it should be num_kv_head.
but it is hardwired to 2.

Signed-off-by: yocox <yocox@nvidia.com>

nvyocox commented:
/bot run

tensorrt-cicd commented:
[PR_Github #30691](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/30691/) [ run ] triggered by Bot. Commit: `66feb41`

nvyocox committed:
[None][feat] Fix pattern match & rewrite for qwen3

Signed-off-by: yocox <yocox@nvidia.com>

tensorrt-cicd commented:
[PR_Github #30956](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/30956/) [ run ] triggered by Bot. Commit: `66feb41`

nvyocox committed:
[None][feat] Fix regression fails

- Remove "arg_" prefix for add_graph_input().
  See https://github.com/NVIDIA/TensorRT-LLM/pull/10117#discussion_r2656862861
  Remove all prefix in the code to fix the regression.

- Replace test_ad_export_onnx model path from "Qwen/Qwen2.5-0.5B"
  to "/home/scratch.trt_llm_data/llm-models/Qwen2.5-0.5B-Instruct"
  to avoid "huggingface_hub.errors.HfHubHTTPError: 429 Client Error: Too Many Requests ..." error.

Signed-off-by: yocox <yocox@nvidia.com>

lucaslie commented on tensorrt_llm/_torch/auto_deploy/transform/interface.py:
```diff
@@ -397,10 +407,26 @@ def __call__(
         return mod
 
+    @final
+    def _visualize_graph(self, mod: nn.Module, idx: int) -> None:
```
we may want to consider doing this for each subgraph. Leaving a note here so that we can address this in https://github.com/NVIDIA/TensorRT-LLM/issues/10203

lucaslie reviewed (approved):
Fantastic!

nvyocox committed:
[None][feat] Update per Lucas's review

- Simplify adapt_to_driveos_llm clean up graph. Remove:

    gm.graph.eliminate_dead_code()
    gm.graph.lint()
    gm.recompile()

  and Use is_clean=False instead.

- Adapt transformations from main:
  - match_bmm_moe_pattern
  - fuse_fp8_moe
  - fuse_nvfp4_moe

- Enhance comment for debug_visualize_dir option

Signed-off-by: yocox <yocox@nvidia.com>

nvyocox commented on tensorrt_llm/_torch/auto_deploy/transform/library/export_to_onnx.py:
```diff
@@ -0,0 +1,418 @@
+        model_dir = factory.model
+        process_chat_template(model_dir, output_dir)
+
+    def _apply(
```
Thank you for the information. It is really useful for me. Because, pohan talked with me about current configuration will have problem to handle multimodal models (and it does) because the model might need to be treated as several parts. After I handle the quantization support, I will go for the VLM, and I may have to distinguish the `_apply` and the `apply_to_full_model`.

nvyocox committed:
[None][feat] Visualize add tensor date type

Signed-off-by: yocox <yocox@nvidia.com>

nvyocox committed:
[None][feat] Fix regression

- Specify model path with get_small_model_config
- Simplify test code, make batch size and seq not argument

Signed-off-by: yocox <yocox@nvidia.com>

tensorrt-cicd commented:
[PR_Github #31680](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/31680/) [ run ] triggered by Bot. Commit: `d42fe2b`

tensorrt-cicd commented:
[PR_Github #31680](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/31680/) [ run ] completed with state `SUCCESS`. Commit: `d42fe2b` 
[/LLM/main/L0_MergeRequest_PR pipeline #24508](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/24508/) completed with status: 'SUCCESS'
Pipeline passed with automatic retried tests. Check the [rerun report](https://urm.nvidia.com/artifactory/sw-tensorrt-generic/llm-artifacts//LLM/main/L0_MergeRequest_PR/24508/test-results/rerun_report.html) for details.

nvyocox commented:
/bot run --stage-list "Build-Docker-Images"

tensorrt-cicd commented:
[PR_Github #32211](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/32211/) [ run ] triggered by Bot. Commit: `d42fe2b`

nvyocox committed:
[None][feat] Rename DriveOS LLM to EdgeLLM

Signed-off-by: yocox <yocox@nvidia.com>

nvyocox commented:
/bot run --disable-fail-fast

nv-guomingz reviewed (approved):
LGTM for doc part

nvluxiaoz commented on tensorrt_llm/_torch/auto_deploy/transform/library/_onnx_schemas.py:
```diff
@@ -0,0 +1,258 @@
+
+
+# ONNX Custom Op Registration for torch_attention
+_torch_attention_schema = defs.OpSchema(
```
Do we need this because TRT OOTB recognize this or something else?

ZhanruiSunCh commented:
/bot run --stage-list "Build-Docker-Images"

tensorrt-cicd commented:
[PR_Github #32852](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/32852/) [ run ] triggered by Bot. Commit: `7a16abb`

nvluxiaoz commented on tests/unittest/_torch/auto_deploy/unit/singlegpu/test_ad_export_onnx.py:
```diff
@@ -0,0 +1,64 @@
+    assert os.path.exists(os.path.join(output_dir, "special_tokens_map.json"))
+    assert os.path.exists(os.path.join(output_dir, "tokenizer.json"))
+    assert os.path.exists(os.path.join(output_dir, "tokenizer_config.json"))
+    assert os.path.exists(os.path.join(output_dir, "vocab.json"))
```
Do you plan to add embedding sharing in this MR or we should make it separate later?

nvyocox commented on docs/source/torch/auto_deploy/advanced/export_onnx.md:
```diff
@@ -0,0 +1,111 @@
+
+| Option | Type | Default | Description |
+|--------|------|---------|-------------|
+| `--model` | str | `TinyLlama/TinyLlama-1.1B-Chat-v1.0` | HuggingFace model name or path to a local checkpoint |
```
Yes, I missed that. Thank you for pointing that out.

nvyocox commented:
/bot run --disable-fail-fast

nvyocox commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #33156](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/33156/) [ run ] triggered by Bot. Commit: `c8258d4`

nvyocox committed:
[None][feat] Update CI docker image

Signed-off-by: yocox <yocox@nvidia.com>

tensorrt-cicd commented:
[PR_Github #33863](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/33863/) [ run ] completed with state `SUCCESS`. Commit: `4384fe5` 
[/LLM/main/L0_MergeRequest_PR pipeline #26113](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/26113/) completed with status: 'FAILURE'

⚠️ **Action Required:**
- Please check the failed tests and fix your PR
- If you cannot view the failures, ask the CI triggerer to share details
- Once fixed, request an NVIDIA team member to trigger CI again

nvyocox commented:
/bot run

nvyocox committed:
[None][feat] Fix CI after rebase per Lucas suggest

Signed-off-by: yocox <yocox@nvidia.com>

tensorrt-cicd commented:
[PR_Github #34175](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/34175/) [ run ] triggered by Bot. Commit: `c90c23b`

tensorrt-cicd commented:
[PR_Github #34175](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/34175/) [ run ] completed with state `SUCCESS`. Commit: `c90c23b` 
[/LLM/main/L0_MergeRequest_PR pipeline #26368](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/26368/) completed with status: 'FAILURE'

⚠️ **Action Required:**
- Please check the failed tests and fix your PR
- If you cannot view the failures, ask the CI triggerer to share details
- Once fixed, request an NVIDIA team member to trigger CI again

lucaslie reviewed (approved):
That was a great last minute addition to add the explicit `export_onnx` function :)

tensorrt-cicd commented:
[PR_Github #34227](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/34227/) [ run ] completed with state `DISABLED`
CI server is currently disabled for scheduled maintenance. Estimated completion time: 8 AM PST on 1/30.

lucaslie commented:
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1" --disable-fail-fast
